File: app/src/main/java/com/hqq/example/demo/FragmentTestActivity.java
package com.hqq.example.demo;

import androidx.fragment.app.Fragment;

import android.app.Activity;
import android.content.Intent;

import com.hqq.core.ui.vm.BaseVmActivity;
import com.hqq.core.ui.vm.BaseViewModel;
import com.hqq.example.R;
import com.hqq.example.databinding.ActivityFragmentTestBinding;

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.example.demo
 * @FileName :   FragmentTestActivity
 * @Date : 2020/7/28 0028  下午 5:51
 * @Email :  qiqiang213@gmail.com
 * @Descrive :
 */
public class FragmentTestActivity extends BaseVmActivity<ActivityFragmentTestBinding, BaseViewModel> {
    public static void open(Activity context) {
        Intent starter = new Intent(context, FragmentTestActivity.class);
        context.startActivityForResult(starter, -1);
    }


    @Override
    public int getLayoutId() {
        return R.layout.activity_fragment_test;
    }

    @Override
    public void initViews() {
        Fragment fragment = TestFragment.newInstance();
        addFragment(fragment, R.id.fl_layout);
    }


    /**
     * 添加 fragment 到 FrameLayout
     *
     * @param fragment fragment
     * @param id       FrameLayout
     */
    @Deprecated
    public void addFragment(Fragment fragment, int id) {
        if (!fragment.isAdded()) {
            getSupportFragmentManager().beginTransaction().add(id, fragment).commit();
        }

    }

    @Override
    public int getBindingViewModelId() {
        return 0;
    }
}

File: app/src/main/java/com/hqq/example/ui/jetpack/databinding/FragmentTestActivity.kt
package com.hqq.example.ui.jetpack.databinding

import androidx.fragment.app.Fragment
import com.hqq.core.ui.base.BaseVmActivity
import com.hqq.core.utils.FragmentUtils
import com.hqq.example.R
import com.hqq.example.databinding.ActivityFragmentTestBinding
import com.hqq.example.demo.TestFragment

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.example.demo
 * @FileName :   FragmentTestActivity
 * @Date : 2020/7/28 0028  下午 5:51
 * @Email :  qiqiang213@gmail.com
 * @Descrive :
 */
class FragmentTestActivity : BaseVmActivity<FragmentTestViewModel, ActivityFragmentTestBinding>() {

    override val layoutId: Int
        get() = R.layout.activity_fragment_test

    override fun initViews() {
        val fragment: Fragment = TestFragment.newInstance()
        FragmentUtils(supportFragmentManager).addOrShowFragment(fragment, R.id.fl_layout)
    }

    override val bindingViewModelId: Int
        get() = 0

}